src/System.Private.CoreLib/shared/System/Environment.cs

```diff
@@ -6,43 +6,107 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Reflection;
-using System.Runtime.CompilerServices;
-using Internal.Runtime.Augments;
 
 namespace System
 {
     public static partial class Environment
     {
+        // If you change this method's signature then you must change the code that calls it
+        // in excep.cpp and probably you will have to visit mscorlib.h to add the new signature
+        // as well as metasig.h to create the new signature type
+        internal static string GetResourceStringLocal(string key) => SR.GetResourceString(key);
+
         public static string GetEnvironmentVariable(string variable)
         {
-            return EnvironmentAugments.GetEnvironmentVariable(variable);
+            if (variable == null)
+                throw new ArgumentNullException(nameof(variable));
+
+            return GetEnvironmentVariableCore(variable);
         }
 
         public static string GetEnvironmentVariable(string variable, EnvironmentVariableTarget target)
         {
-            return EnvironmentAugments.GetEnvironmentVariable(variable, target);
+            if (target == EnvironmentVariableTarget.Process)
+                return GetEnvironmentVariable(variable);
+
+            if (variable == null)
+                throw new ArgumentNullException(nameof(variable));
+
+            bool fromMachine = ValidateAndConvertRegistryTarget(target);
+            return GetEnvironmentVariableFromRegistry(variable, fromMachine);
         }
 
-        public static IDictionary GetEnvironmentVariables()
+        public static IDictionary GetEnvironmentVariables() =>
+            ToHashtable(EnumerateEnvironmentVariables(GetEnvironmentCharArray())); // returns a Hashtable for back compat
+
+        public static IDictionary GetEnvironmentVariables(EnvironmentVariableTarget target)
         {
-            // To maintain complete compatibility with prior versions we need to return a Hashtable.
-            // We did ship a prior version of Core with LowLevelDictionary, which does iterate the
-            // same (e.g. yields DictionaryEntry), but it is not a public type.
-            //
-            // While we could pass Hashtable back from CoreCLR the type is also defined here. We only
-            // want to surface the local Hashtable.
-            return EnvironmentAugments.EnumerateEnvironmentVariables().ToHashtable();
+            if (target == EnvironmentVariableTarget.Process)
+                return GetEnvironmentVariables();
+
+            bool fromMachine = ValidateAndConvertRegistryTarget(target);
+            return ToHashtable(EnumerateEnvironmentVariablesFromRegistry(fromMachine));
         }
 
-        public static IDictionary GetEnvironmentVariables(EnvironmentVariableTarget target)
+        private static IEnumerable<KeyValuePair<string, string>> EnumerateEnvironmentVariables(char[] block)
```

Comment:
This extra `IEnumerable` layer is a left-over from times when HashTable did not live in CoreLib. It is unnecessary now and it is also a reason why we allocate copy of the environment in `char[]` from `char*` - C# does not let you use unsafe code in enumerators.